tracer/src/Datadog.Trace/IAST/StackWalker.cs

```diff
@@ -0,0 +1,35 @@
+// <copyright file="StackWalker.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System.Diagnostics;
+using System.Linq;
+
+namespace Datadog.Trace.Iast
+{
+    internal static class StackWalker
+    {
+        public static readonly string[] AssemblyNamesToSkip = { "Datadog.Trace", "System.Security.Cryptography.Primitives" };
+
+        private const int DefaultSkipFrames = 2;
+
+        public static StackFrame? GetFrame()
+        {
+            var stackTrace = new StackTrace(DefaultSkipFrames, true);
+
+            foreach (var frame in stackTrace.GetFrames())
+            {
+                var assembly = frame?.GetMethod()?.DeclaringType?.Assembly?.GetName()?.Name;
```

Comment:
From the MSDN docs, it looks look some of these can never return null.
```suggestion
                var assembly = frame.GetMethod()?.DeclaringType?.Assembly.GetName().Name;
```

Reply:
But I wouldn't bother if this is not on the hot path :)